rfqmsg/custom_channel_data.go

```diff
@@ -43,6 +48,26 @@ type JsonAssetChannel struct {
 	IncomingHtlcBalance uint64             `json:"incoming_htlc_balance"`
 }
 
+// HasAllAssetIDs checks if the OpenChannel contains all asset IDs in the
+// provided set. It returns true if all asset IDs are present, false otherwise.
+func (c *JsonAssetChannel) HasAllAssetIDs(ids fn.Set[asset.ID]) bool {
+	for id := range ids {
+		found := false
+		for _, fundingAsset := range c.FundingAssets {
+			if fundingAsset.AssetGenesis.AssetID == id.String() {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
Could also construct two sets of id strings and then diff or union and length check.